packages/fuel-indexer-lib/src/manifest.rs

```diff
@@ -46,6 +47,30 @@ impl ToString for Module {
     }
 }
 
+/// Represents contract IDs in a `Manifest` struct.
+///
+/// - `Single`: Represents an optional single contract ID as an `Option<String>`.
+/// - `Multiple`: Represents a vector of optional contract IDs as a `Vec<Option<String>>`.
+#[derive(Debug, Deserialize, Serialize, Clone)]
+pub enum ContractIds {
+    Single(Option<String>),
+    Multiple(Vec<Option<String>>),
+}
+
+impl FromStr for ContractIds {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if s.starts_with('[') {
+            serde_json::from_str::<Vec<Option<String>>>(s)
```

Comment:
@rvmelkonian Curious as to why this is `Vec<Option<String>>` and not `Vec<String>`. Under what circumstances would any of these contract IDs be `None`?

Reply:
Updated to: 
```
pub enum ContractIds {
    ///Single represents a single contract ID as an `Option<String>`.
    Single(Option<String>),
    ///Multiple represents a vector of contracts IDs as a Vec<String>.
    Multiple(Vec<String>),
}
```
Single still should be Option<String> as before, so that we can deal with ~ yaml values.